src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+
+    Herbivore.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor() {
```

Comment:
[FUNCTION NAMING] - The constructor for `Carnivore` should accept a `name` argument and pass it to the superclass constructor with `super(name)`. Currently, it does not take any parameters, which means that all carnivores will have no name.